src/interpreter/mod.rs

```diff
@@ -243,8 +243,11 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
             Str(ref s) => {
                 let ptr = self.memory.allocate(s.len(), 1)?;
-                self.memory.write_bytes(ptr, s.as_bytes())?;
-                self.memory.freeze(ptr.alloc_id)?;
+                // don't write to the zst allocation
```

Comment:
I would prefer if this code didn't have to change. `freeze` already special cases the ZST allocation, and writing zero bytes shouldn't be a problem.